covalent/_shared_files/utils.py

```diff
@@ -221,5 +221,22 @@ def get_named_params(func, args, kwargs):
     return (named_args, named_kwargs)
 
 
-# Dictionary to map Dask clients to their scheduler addresses
-_address_client_mapper = {}
+def format_server_url(hostname: str = None, port: int = None) -> str:
+    if hostname is None:
+        hostname = get_config("dispatcher.address")
+    if port is None:
+        port = int(get_config("dispatcher.port"))
+
+    url = hostname
+    if not url.startswith("http"):
+        if port == 443:
+            url = f"https://{url}"
+        else:
+            url = f"http://{url}"
+
+    # Inject port
+    if port not in [80, 443]:
+        parts = url.split("/")
+        url = "".join(["/".join(parts[:3])] + [f":{port}/"] + ["/".join(parts[3:])])
```

Comment:
Not sure what this logic is doing? And the ports are hard-coded. Ideally, would it be read from some config file?

Reply:
@wjcunningham7 this is yours : )